saleor/graphql/order/mutations/draft_orders.py

```diff
@@ -397,9 +397,10 @@ def perform_mutation(cls, _root, info, id):
                 line_data = OrderLineData(
                     line=line, quantity=line.quantity, variant=line.variant
                 )
+                channel_slug = order.channel.slug
                 try:
-                    allocate_stocks([line_data], country, order.channel.slug)
-                    allocate_preorders([line_data], order.channel.slug)
+                    allocate_stocks([line_data], country, channel_slug)
```

Comment:
Shouldn't be wrapped in transaction? What should happen with allocated_stocks when allocate_preorders will raise an error?

Reply:
Right, added.